esp-hal-embassy/build.rs

```diff
@@ -39,16 +39,63 @@ fn main() -> Result<(), Box<dyn Error>> {
     config.define_symbols();
 
     // emit config
-    generate_config(
+    let crate_config = generate_config(
         "esp_hal_embassy",
         &[(
             "low-power-wait",
             "Enables the lower-power wait if no tasks are ready to run on the thread-mode executor. This allows the MCU to use less power if the workload allows. Recommended for battery-powered systems. May impact analog performance.",
             Value::Bool(true),
             None
-        )],
+        ),
+        (
+            "generic-queue-size",
+            "The size of the generic queue. Only used if `generic-queue` is enabled.",
+            Value::Integer(64),
+            Some(Validator::PositiveInteger),
+        ),
+        (
+            "timer-queue",
+            "The flavour of the timer queue provided by this crate. Accepts one of 'single-integrated', 'multiple-integrated' or 'generic'. Integrated queues require the 'executors' feature to be enabled.",
+            Value::String(if cfg!(feature = "executors") {
+                String::from("single-integrated")
+            } else {
+                String::from("generic")
+            }),
+            Some(Validator::Custom(Box::new(|value| {
+               let Value::String(string) = value else {
+                     return Err(Error::Validation(String::from("Expected a string")));
+               };
+
+                match string.as_str() {
+                    "single-integrated" => Ok(()), // preferred for ease of use
+                    "multiple-integrated" => Ok(()), // preferred for performance
+                    "generic" => Ok(()), // allows using embassy-time without the embassy executors
+                    _ => Err(Error::Validation(format!("Expected 'single-integrated', 'multiple-integrated' or 'generic', found {string}")))
+                }
+            })))
+        )
+    ],
         true,
     );
 
+    println!("cargo:rustc-check-cfg=cfg(integrated_timers)");
+    println!("cargo:rustc-check-cfg=cfg(single_queue)");
+    println!("cargo:rustc-check-cfg=cfg(generic_timers)");
+
+    match &crate_config["ESP_HAL_EMBASSY_TIMER_QUEUE"] {
```

Comment:
We need to figure out passing config options to tests for this. We should also figure out testing multiple feature sets, or configs in this case.